### PR TITLE
Remove register methods

### PR DESCRIPTION
**Pull request by PoignardAzur**
Replace with `Widget::accepts_xxx()` methods
Tweak accessibility code
Remove `BuildFocusChain` event

Remove `WidgetState::is_portal`, `register_as_portal()`, and overall remove the idea of a first-class "portal" concept.

### CODE REVIEW

**Comment by PoignardAzur** [comment]
Man.

I just spend half an hour debugging an error I would have found immediately if I'd looked at the clippy error log. I feel like an idiot.

**Review by jaredoconnell — APPROVED** [review]
I tested a lot of examples for this one and cannot find any regressions.

File: masonry/src/passes/update.rs [hunk 1]
```diff
@@ -620,16 +623,9 @@ fn update_focus_chain_for_widget(
...
-    {
-        let mut ctx = LifeCycleCtx {
-            global_state,
-            widget_state: state.item,
-            widget_state_children: state.children.reborrow_mut(),
-            widget_children: widget.children.reborrow_mut(),
-        };
-        widget.item.lifecycle(&mut ctx, &LifeCycle::BuildFocusChain);
+    if state.item.accepts_focus {
+        state.item.focus_chain.push(id);
```
Comment (DJMcNab) [inline masonry/src/passes/update.rs:628]: Why don't we just extend the global focus chain here? That is, what does the per-widget focus chain achive, now that we have an explicit state for it?

File: masonry/src/widget/label.rs [hunk 2]
```diff
@@ -55,14 +55,15 @@ impl Label {
...
-            skip_pointer: false,
+            interactive: true,
         }
     }
 
-    // TODO - Rename
-    // TODO - Document
-    pub fn with_skip_pointer(mut self, skip_pointer: bool) -> Self {
-        self.skip_pointer = skip_pointer;
+    /// Sets the value returned by [`accepts_pointer_interaction`].
```
Comment (DJMcNab) [inline masonry/src/widget/label.rs:62]: Hmm, I'm surprised that this doc resolves? Is there a rustdoc feature I didn't know about (automatic handling of trait methods?)

It isn't shown as a valid link by rust-analyzer. I've not tried it in rustdoc

File: masonry/src/passes/update.rs (hunk 1 above)
Comment (PoignardAzur) [reply masonry/src/passes/update.rs:628]: It does some caching.

It's not ideal, though. I think the whole "focus chain" concept should be refactored.

File: masonry/src/widget/label.rs (hunk 2 above)
Comment (PoignardAzur) [reply masonry/src/widget/label.rs:62]: Nope, rustdoc doesn't resolve it either.

File: masonry/src/passes/update.rs (hunk 1 above)
Comment (DJMcNab) [reply masonry/src/passes/update.rs:628]: Oh of course, some children won't actually have this pass run on them?

Comment (PoignardAzur) [reply masonry/src/passes/update.rs:628]: Yeah, the pass doesn't run on the entire tree.

**Pull request merged**